x-pack/plugins/security_solution/server/lib/detection_engine/signals/wrap_hits_factory.ts

```diff
@@ -25,11 +25,15 @@ export const wrapHitsFactory = ({
   const wrappedDocs: WrappedSignalHit[] = events.flatMap((doc) => [
     {
       _index: signalsIndex,
-      // TODO: bring back doc._version
-      _id: generateId(doc._index, doc._id, '', ruleSO.attributes.params.ruleId ?? ''),
+      _id: generateId(
+        doc._index,
+        doc._id,
+        doc._version ? doc._version.toString() : '',
+        ruleSO.attributes.params.ruleId ?? ''
+      ),
```

Comment:
Is it possible to type the line below in the file above and then we avoid this as cast?

```ts
_source: buildBulkBody(ruleSO, doc as SignalSourceHit),
```